services/scabbard/libscabbard/src/service/factory.rs

```diff
@@ -156,15 +156,25 @@ impl ServiceArgValidator for ScabbardArgValidator {
         let admin_keys_str = args
             .get("admin_keys")
             .ok_or_else(|| ServiceArgValidationError("admin_keys argument not provided".into()))?;
-        let admin_keys: Vec<String> = serde_json::from_str(admin_keys_str).map_err(|err| {
-            ServiceArgValidationError(format!("failed to parse admin_keys list: {}", err,))
-        })?;
+
+        let admin_keys = {
+            if admin_keys_str.starts_with('[') {
+                serde_json::from_str(admin_keys_str).map_err(|err| {
+                    ServiceArgValidationError(format!("failed to parse admin_keys list: {}", err,))
+                })?
+            } else {
+                admin_keys_str
+                    .split(',')
+                    .map(String::from)
+                    .collect::<Vec<String>>()
+            }
+        };
 
         for key in admin_keys {
-            let key_bytes = parse_hex(&key).map_err(|_| {
+            let key_bytes = parse_hex(&key).map_err(|err| {
                 ServiceArgValidationError(format!(
-                    "{} is not a valid hex-formatted public key",
-                    key
+                    "{:?} is not a valid hex-formatted public key {}",
+                    key, err
```

Comment:
Is err relevant here? If it is, should there be a : before {}?

Reply:
I actually meant to remove this. I added it when I was debugging. I will remove it.

Reply:
fixed